src/test/java/seedu/momentum/model/ProjectBookWithUiTest.java

```diff
@@ -0,0 +1,118 @@
+                        Model.PREDICATE_SHOW_ALL_TRACKED_ITEMS, NAME_COMPARE, false);
+        assertEquals(false, projectBookTagsInvisible.isTagsVisible());
+    }
+
+    @Test
+    public void equals() {
+        // same object -> returns true
+        assertTrue(PROJECT_BOOK_WITH_UI.equals(PROJECT_BOOK_WITH_UI));
+
+        // different types -> returns false
+        assertFalse(PROJECT_BOOK_WITH_UI.equals("1"));
+
+        // null -> return false
+        assertFalse(PROJECT_BOOK_WITH_UI.equals(null));
+
+        //same viewMode, project, predicate, comparator -> return true
+        ProjectBookWithUi sameProjectbookWithUi = new ProjectBookWithUi(new ProjectBook(), ViewMode.TASKS,
+                VALID_PROJECT, Model.PREDICATE_SHOW_ALL_TRACKED_ITEMS, NAME_COMPARE, true);
+        assertTrue(PROJECT_BOOK_WITH_UI.equals(sameProjectbookWithUi));
+
+        //same viewMode, both projects null, same predicate, same comparator -> return true
+        ProjectBookWithUi nullProjectBookWithUi =
+                new ProjectBookWithUi(new ProjectBook(), ViewMode.TASKS, null,
+                        Model.PREDICATE_SHOW_ALL_TRACKED_ITEMS, NAME_COMPARE, true);
+        sameProjectbookWithUi = new ProjectBookWithUi(new ProjectBook(), ViewMode.TASKS,
+                null, Model.PREDICATE_SHOW_ALL_TRACKED_ITEMS, NAME_COMPARE, true);
+        assertTrue(nullProjectBookWithUi.equals(sameProjectbookWithUi));
+
+        // different viewMode -> return false
+        ProjectBookWithUi differentProjectbookWithUi = new ProjectBookWithUi(new ProjectBook(), ViewMode.PROJECTS,
+                VALID_PROJECT, Model.PREDICATE_SHOW_ALL_TRACKED_ITEMS, NAME_COMPARE, true);
+        assertFalse(PROJECT_BOOK_WITH_UI.equals(differentProjectbookWithUi));
+
+        // different project -> returns false
+        Project otherProject = new ProjectBuilder().withName("OTHER").build();
+        differentProjectbookWithUi = new ProjectBookWithUi(new ProjectBook(), ViewMode.TASKS,
+                otherProject, Model.PREDICATE_SHOW_ALL_TRACKED_ITEMS, NAME_COMPARE, true);
+        assertFalse(PROJECT_BOOK_WITH_UI.equals(differentProjectbookWithUi));
+
+        // different tags visibility -> returns false
+        differentProjectbookWithUi = new ProjectBookWithUi(new ProjectBook(), ViewMode.TASKS,
+                VALID_PROJECT, Model.PREDICATE_SHOW_ALL_TRACKED_ITEMS, NAME_COMPARE, false);
+        assertFalse(PROJECT_BOOK_WITH_UI.equals(differentProjectbookWithUi));
+
+        //        // different predicate -> returns false
+        //        Predicate<TrackedItem> completePredicate = new CompletionStatusPredicate(Arrays.asList("incomplete"));
+        //        differentProjectbookWithUi = new ProjectBookWithUi(new ProjectBook(), ViewMode.TASKS,
+        //                otherProject, completePredicate, nameCompare);
+        //        assertFalse(projectBookWithUi.equals(differentProjectbookWithUi));
```

Comment:
Can this be removed?

Reply:
This is related to the first comment, it is supposed to be there but it is conflicting with the other tests. Will clarify and resolve later when we call to fix code quality etc.